### PR TITLE
Fix assertion failure for optional VB attribute with invalid type (Approach 2)

### PR DESCRIPTION
**Pull request by Youssef1313**
A different fix for https://github.com/dotnet/roslyn/pull/60924

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@Youssef1313 It doesn't look like this PR is fixing any specific recorded issue and the description provides no details about the problem.

**Comment by Youssef1313** [comment]
@AlekseyTs See https://github.com/dotnet/roslyn/pull/60924#issuecomment-1112547488

File: src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb [hunk 1]
```diff
@@ -3404,11 +3404,15 @@ Bailout:
                 Dim passedExpression As BoundExpression = binder.PassArgumentByVal(argument, conv, targetType, BindingDiagnosticBag.Discarded)
 
                 If Not passedExpression.IsConstant Then ' Trying to match native compiler behavior in Semantics::IsValidAttributeConstant
-                    Dim visitor As New Binder.AttributeExpressionVisitor(binder, passedExpression.HasErrors)
-                    visitor.VisitExpression(passedExpression, BindingDiagnosticBag.Discarded)
-
-                    If visitor.HasErrors Then
+                    If Not targetType.IsValidTypeForAttributeArgument(binder.Compilation) Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb:3407]: >Not targetType.IsValidTypeForAttributeArgument(binder.Compilation) [](http://example.com/codeflow?start=23&length=66)

It looks like there is a t least one more place where ```AttributeExpressionVisitor``` is used and that place is not "protected". I think fixing the visitor is a better approach.

**Comment by Youssef1313** [comment]
Closing in favor of #60924

**Pull request closed without merging**